core/DataAccess/Model.php

```diff
@@ -304,11 +304,12 @@ public function insertRecord($tableName, $fields, $record, $name, $value)
     {
         // duplicate idarchives are Ignored, see https://github.com/piwik/piwik/issues/987
         $query = "INSERT IGNORE INTO " . $tableName . " (" . implode(", ", $fields) . ")
-                  VALUES (?,?,?,?,?,?,?,?)";
+                  VALUES (?,?,?,?,?,?,?,?) ON DUPLICATE KEY UPDATE " . end($fields) . " = ?";
```

Comment:
using end() should work most of the time but to be safe, let's ensure we use correct field.

Also is it do-able adding a test for this?